internal/pkg/cli/pipeline_init.go

```diff
@@ -326,38 +392,62 @@ func (o *initPipelineOpts) parseBitbucketRepoDetails() error {
 	o.repoName = repoDetails.name
 	o.repoOwner = repoDetails.owner
 
-	if o.repoBranch == "" {
-		o.repoBranch = defaultBBBranch
-	}
 	return nil
 }
 
 func (o *initPipelineOpts) selectURL() error {
-	// Fetches and parses all remote repositories.
-	err := o.runner.Run("git", []string{"remote", "-v"}, exec.Stdout(&o.buffer))
+	repos, err := o.fetchAndParseURLs()
 	if err != nil {
-		return fmt.Errorf("get remote repository info: %w; make sure you have installed Git and are in a Git repository", err)
+		return fmt.Errorf("fetch and parse URLs: %w", err)
 	}
-	urls, err := o.parseGitRemoteResult(strings.TrimSpace(o.buffer.String()))
-	if err != nil {
-		return err
+
+	var formattedRepos []string
+	for url, name := range repos {
+		formattedRepos = append(formattedRepos, fmt.Sprintf("%s: %s", name, url))
 	}
-	o.buffer.Reset()
 
 	// Prompts user to select a repo URL.
 	url, err := o.prompt.SelectOne(
 		pipelineSelectURLPrompt,
 		pipelineSelectURLHelpPrompt,
-		urls,
-		prompt.WithFinalMessage("Repository URL:"),
+		formattedRepos,
+		prompt.WithFinalMessage("Repository:"),
 	)
 	if err != nil {
 		return fmt.Errorf("select URL: %w", err)
 	}
-	if err := o.validateURL(url); err != nil {
+	repoParts := strings.Split(url, ": ")
+	if err := o.validateURLType(repoParts[1]); err != nil {
```

Comment:
Do we need to validate for the existence as well, just like in `Validate()`?

Reply:
Oh, you mean confirm that the URL is a git remote? The options that this URL was selected from were generated by the `git remote -v` command, so this URL has to exist. 
Does that answer your question?

Reply:
right right! Sorry it slipped my mind!